astroplan/scheduling.py

```diff
@@ -90,6 +92,81 @@ def from_exposures(cls, target, priority, time_per_exposure,
         return ob
 
 
+class Scorer(object):
+    """
+    Returns scores and score arrays from the evaluation of constraints on
+    observing blocks
+    """
+    def __init__(self, blocks, observer, schedule):
+        """
+
+        Parameters
+        ----------
+        blocks : list of `~astroplan.scheduling.ObservingBlock` objects
+            list of blocks that need to be scored
+        observer : `~astroplan.Observer`
+            the observer
+        schedule : `~astroplan.scheduling.Schedule`
+            The schedule inside which the blocks should fit
+        """
+        self.blocks = blocks
+        self.observer = observer
+        self.schedule = schedule
+        # the ranges below should change if there are new constraints made
+        self.constraint_list = [eval(name) for name in
+                                astroplan.constraints.__all__[:-6]
+                                if 'Constraint' in name]
+        self.scheduling_constraints = [eval(name) for name in
+                                       astroplan.constraints.__all__[-5:]
+                                       if 'Constraint' in name]
```

Comment:
There are a few ways this can be resolved: have only global constraints be vectorized across targets a la #190 (which could work, but would require care in the scheduler to retain the order of `ObservingBlocks` compared to the vectorized `Constraints`); have it evaluate each target individually (this eliminates any use of the optimizations in vectorizing the constraints); or have a list of all the `Constraints` defined in `scheduling.py` or 'constraints.py`. All three of the solutions above solve the import issue above as a by-product. I will try to see if I can implement the third solution, as it preserves most of the potential optimisation, and fall back on the other two if I run into too many issues.